src/lib/rest/rest.cpp

```diff
@@ -1072,6 +1101,18 @@ static int connectionTreat
     LM_T(LmtUriParams, ("'default' value for notifyFormat (ciP->outFormat == %d)): '%s'", ciP->outFormat, ciP->uriParam[URI_PARAM_NOTIFY_FORMAT].c_str()));
   }
 
+  //
+  // Here, if the incoming request was too big, return error abouyt it
+  //
+  if (ciP->httpHeaders.contentLength > PAYLOAD_MAX_SIZE)
```

Comment:
Or maybe the MHD detect the Content-Length vs. actual payload disalignment and send the error itself... We need to test what hapepend with a crafted HTTP request (maybe curl can do that).